CenterForOpenScience/osf.io
# [Py3][ENG-129] Even more Py3 backwards compatibility changes

## Purpose

Get code in line before we switch over to Python 3.

## Changes

- Removed lambda parenthesis 
- change relative imports to absolutes
- moved urllib library to future
- changed the http status library to the DRF one which is good in both 2 and 3

## QA Notes

No user facing changes
Check links of addon urls to see if there are any changes in logging and display. Check displayed links and connect/disconnect from Gitlab, Bitbucket and Github.

## Documentation

Tech task, no docs

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-150
https://openscience.atlassian.net/browse/ENG-129
